src/history/hs.c

```diff
@@ -799,27 +799,6 @@ __wt_hs_insert_updates(WT_SESSION_IMPL *session, WT_PAGE *page, WT_MULTI *multi)
                 continue;
 
             non_aborted_upd = upd;
-
-            /* If we've seen a smaller timestamp before, use that instead. */
-            if (min_insert_ts < upd->start_ts) {
-                /*
-                 * Resolved prepared updates will lose their durable timestamp here. This is a
-                 * wrinkle in our handling of out-of-order updates.
-                 */
-                if (upd->start_ts != upd->durable_ts) {
-                    WT_ASSERT(session, min_insert_ts < upd->durable_ts);
-                    WT_STAT_CONN_INCR(session, cache_hs_order_lose_durable_timestamp);
-                }
-                __wt_verbose(session, WT_VERB_TIMESTAMP,
-                  "fixing out-of-order updates during insertion; start_ts=%s, durable_start_ts=%s, "
-                  "min_insert_ts=%s",
-                  __wt_timestamp_to_string(upd->start_ts, ts_string[0]),
-                  __wt_timestamp_to_string(upd->durable_ts, ts_string[1]),
-                  __wt_timestamp_to_string(min_insert_ts, ts_string[2]));
-                upd->start_ts = upd->durable_ts = min_insert_ts;
-                WT_STAT_CONN_INCR(session, cache_hs_order_fixup_insert);
-            } else
-                min_insert_ts = upd->start_ts;
```

Comment:
Would a different approach to this change be to only save `min_insert_ts` if `upd->start_ts` is not `WT_TS_NONE`?
Is that what this change is about? We don't want to reset timestamps in the past in the special case of mixed-mode timestamps? I think it's worrying that something is relying on that semantic. Does it mean that a transaction-ID based check is causing the globally visible update to be skipped, then relying on a timestamp check to skip seeing the first update that was older than it?

Reply:
I'm discussing with Vamsi about a better solution for this problem. This PR actually doesn't handle all the edge cases and I think there are better ways.

Reply:
The problem is previously we skipping inserting updates after a mixed mode operation but WT-6388 is changing the timestamps after the mixed mode operation to zero so we can't detect mixed mode operation anymore, thus trigger the assert.

Reply:
The best solution I have in mind is to let obsolete check tell us the first globally visible update and skip inserting everything after that.